### ISSUE TITLE
[Question] How to add .ascx item type?

### ISSUE THREAD

**Issue by dazinator** [Microsoft/VSProjectSystem#91]
For my project system, i'd like to allow "ascx" files to be visible in the "Add New Item" window.

I am just getting started, and have so far enabled certain items to show up using project capabilities, such as "javascript" - but there doesn't appear to be a capability to allow ASP.NET 4 webforms content to show up, such as ASPX, ASCX etc etc - So I would like to know the steps necessary to enable a particular file type to show up, so I can include the ones i need on a case by case basis!

I read the section about adding a new file type - i.e a .foo file - however this is not a new file type, its an existing recognised file type I want to include.

Any help would be much appreciated. Thanks

**Comment by dazinator** [comment]
Think I have made some headway, Correct me If I am wrong, but I think I need to create or find the existing project item templates for asp.net 4's project items, and then include them in my vsix package, but with the <AppliesTo> tag, pointing to my unique projects capability.

**Comment by dazinator** [comment]
Ok, creating my own item template projects, and including them in the vsix works ok - although having to create copies of the existing asp.net 4 item templates is a bit of a pain. Luckily I only need a few so not a big deal for me at present.

**Comment by aienabled** [comment]
I was required to create item templates for XAML pages as they're not provided by default. It's a proper way of doing this currently (discussed in chat https://gitter.im/Microsoft/VSProjectSystem ). I wish someday there will be a repository with reference item templates.

**Comment by dazinator** [comment]
Thanks again - yeah I agree, a repository for these standard templates would be desirable!

**Comment by dazinator** [comment]
Have re-opened this issue, because even though its possible to create an item template that brings in the ascx, code behind, and designer files, by creating a vstemplate like this:

``` xml

 <TemplateContent>
    <References>
      <Reference>
        <Assembly>System</Assembly>
        <Assembly>System.Web</Assembly>
      </Reference>
    </References>

    <ProjectItem SubType="" TargetFileName="$fileinputname$.ascx" ReplaceParameters="true">View.ascx</ProjectItem>
    <ProjectItem SubType="ASPXCodeBehind" TargetFileName="$fileinputname$.ascx.cs" ReplaceParameters="true">View.ascx.cs</ProjectItem>
    <ProjectItem SubType="" TargetFileName="$fileinputname$.ascx.designer.cs" ReplaceParameters="true">View.ascx.designer.cs</ProjectItem>   
  
  </TemplateContent>  

```

This won't by default make the ascx, code behind, and designer files behave like they do in a web project. For example, changing the ascx markup and saving, wont appear to re-generate the designer file.

Also, it's not clear on what "rules" are required to make the CPS project system recognise and structure these files appropriately. Everything I have tried leads to strange errors like #102 

I'd be greatful to any better men / women than I, that could provide some working example of a CPS project with working ASCX files!

**Comment by dazinator** [comment]
Closed again as this is better covered by #105
